### PR TITLE
feat: added fade-in-out feature to scroll button

### LINKED ISSUE #66: 🚀 Feature: Have ScrollButton fade in and out
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Following up on #50: right now the `ScrollButton` pops in and out of appearance by switching from `display: none` to `display: flex`. It would be nice if it instead faded in and out. Maybe `transition: opacity 350ms` or something like that?

### Additional Info

Note that the button should _both_ have `opacity` and `display` changed. If it's just `opacity: 0` and `display: flex`, users might be able to click on it even though it's not visible.

### PR DESCRIPTION
**Pull request by RuthwikReddy09**
## PR Checklist

- [x] Addresses an existing open issue: fixes #66 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/tidelift-me-up-site/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Added fade-in and fade-out feature to the scroll to top button
Here is the article that i used for reference : https://greywyvern.com/337#

### CODE REVIEW

File: src/components/ScrollButton.module.css [hunk 1]
```diff
@@ -16,11 +16,15 @@
 	text-decoration: none;
 	text-shadow: 0 0 5px var(--color-shadow);
 	z-index: 1;
-	display: none;
+	visibility: hidden;
+	opacity: 0;
+	transition: visibility 0s linear 0.35s, opacity 0.35s linear;
```
Comment (JoshuaKGoldberg) [inline src/components/ScrollButton.module.css]: I tried this out and personally found `0.35s` to be kind of annoyingly slow 😅 sorry for the bad suggestion. What do you think about speeding it up to maybe... 0.2s?

```suggestion
	transition: visibility 0s linear 0.2s, opacity 0.2s linear;
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
This is great, thanks @RuthwikReddy09! 🚀 

The code changes are clean, no complaints. Just the one question about timing. LMK what you think!

File: src/components/ScrollButton.module.css (hunk 1 above)
Comment (RuthwikReddy09) [reply src/components/ScrollButton.module.css]: Yeah sure. I'll make the changes :)

Comment (RuthwikReddy09) [reply src/components/ScrollButton.module.css]: Done. Could you review it please.

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 thanks!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @joshuakgoldberg for maintenance.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Comment by JoshuaKGoldberg** [comment]
@allcontributors please add @RuthwikReddy09 for code

**Pull request merged**